llmfoundry/callbacks/hf_checkpointer.py

```diff
@@ -220,7 +220,18 @@ def _save_checkpoint(self, state: State, logger: Logger):
                 # TODO: after torch 2.1, we can load a state dict into a meta model
                 # and skip the extra model init
                 log.debug(f'Creating new model instance')
-                new_model_instance = type(original_model)(copied_config)
+
```

Comment:
To avoid messing with the model on state, which is sharded and might not save correctly. tbf it might have worked, but I wanted to avoid relying on the HF save_pretrained code meshing with our FSDP context manager for getting the full state dict.

Reply:
I think it might work calling `save_pretrained(..., is_main_process=dist.get_global_rank()==0, ...)` inside the FSDP context manager, but I try to avoid relying on Hugging Face internals wherever possible.

Reply:
resolving, lmk if you disagree